File: baekjoon/2018.cpp
#include <iostream>
using namespace std;

int main()
{
	int n;
	cin >> n;

	int end = 1;
	int total = 0;
	int cnt = 0;
	
	for (int start = 1; start <= n; start++) {
		while (total < n && end <= n) {
			total += end;
			end++;
		}
		if (total == n)
			cnt++;
		total -= start;
	}
	
	cout << cnt;
}

File: baekjoon/2018.py
n = int(input())

end = 1
cnt = 0
total = 0

for start in range(1, n+1):
	while total < n and end <= n:
		total += end
		end += 1
	if total == n:
		cnt += 1
	total -= start
print(cnt)

